### PR TITLE
Fix handling of `--skip-external-dependencies` parameter in `tuist graph` command when `--format json` is specified

### PR DESCRIPTION
**Pull request by GermanVelibekovHouzz**
Resolves https://github.com/tuist/tuist/issues/4326

### Short description 📝

- Applies filter logic passed as arguments to `tuist graph` command when generating for JSON format.
- Removes the dot at the end of `Graph exported to …` for an easier copy and paste of a path.

### How to test the changes locally 🧐

Run `tuist graph <targets> -f json [-d]` for the project generating multiple .xcprojects. The generated JSON file should now only include:
- projects  containing `<targets>`
- projects containing transitive closure dependencies considering additional optional filtering parameters

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, whenever it should be included in the “Added”, “Fixed” or “Changed” section of the CHANGELOG. Note: when included in the CHANGELOG, the title of the PR will be used as entry, please make sure it is clear and suitable.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.
- [ ] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: Sources/TuistKit/Services/GraphService.swift [hunk 1]
```diff
@@ -142,12 +150,18 @@ private enum GraphServiceError: FatalError {
...
+    fileprivate static func from(
+        graph: TuistGraph.Graph,
+        targetsAndDependencies: [GraphTarget: Set<GraphDependency>]
+    ) -> ProjectAutomation.Graph {
+        // generate targets projects only
+        let projects = targetsAndDependencies
+            .map(\.key.project)
+            .reduce(
+                into: [String: ProjectAutomation.Project]()
+            ) {
```
Comment (danyf90) [inline Sources/TuistKit/Services/GraphService.swift:162]:
```suggestion
            .reduce(into: [String: ProjectAutomation.Project]()) {
```

**Review by danyf90 — APPROVED** [review]

File: Sources/TuistKit/Services/GraphService.swift [hunk 2]
```diff
@@ -54,22 +54,30 @@ final class GraphService {
             try FileHandler.shared.delete(filePath)
         }
 
+        let filteredTargetsAndDependencies = graphVizMapper.filter(
+            graph: graph,
+            skipTestTargets: skipTestTargets,
+            skipExternalDependencies: skipExternalDependencies,
+            targetsToFilter: targetsToFilter
+        )
```
[the remark this answers is not in the document]
Comment (GermanVelibekovHouzz) [reply Sources/TuistKit/Services/GraphService.swift]: @danyf90 I see filtering has something to do with `GraphToGraphVizMapper` as its tests are now failing. I suspect this functionality was intended to be there in the first place.

[the remark this answers is not in the document]
Comment (GermanVelibekovHouzz) [reply Sources/TuistKit/Services/GraphService.swift]: Should I introduce the filter method back to the `GraphToGraphVizMapper`?

[the remark this answers is not in the document]
Comment (danyf90) [reply Sources/TuistKit/Services/GraphService.swift]: I guess you should remove filter testing from there and move it to the graphservicetests?

[the remark this answers is not in the document]
Comment (GermanVelibekovHouzz) [reply Sources/TuistKit/Services/GraphService.swift]: I set the extension inside _TuistGenerator/Extensions_. Will it make sense?

**Review by fortmarek — APPROVED** [review]

**Pull request merged**